src/ui/data-panel.ts

```diff
@@ -0,0 +1,372 @@
+import { BooleanInput, Container, Label, Panel, SelectInput } from 'pcui';
+import { Events } from '../events';
+import { Splat } from '../splat';
+import { Histogram } from './histogram';
+import { State } from '../edit-ops';
+import { rgb2hsv } from './color';
+
+const SH_C0 = 0.28209479177387814;
+
+const identity = (v: number) => v;
+const scaleFunc = (v: number) => Math.exp(v);
+const colorFunc = (v: number) => 0.5 + v * SH_C0;
+const sigmoid = (v: number) => {
+    if (v > 0) {
+        return 1 / (1 + Math.exp(-v));
+    }
+
+    const t = Math.exp(v);
+    return t / (1 + t);
+};
+
+const dataFuncs = {
+    x: identity,
+    y: identity,
+    z: identity,
+    scale_0: scaleFunc,
+    scale_1: scaleFunc,
+    scale_2: scaleFunc,
+    f_dc_0: colorFunc,
+    f_dc_1: colorFunc,
+    f_dc_2: colorFunc,
+    opacity: sigmoid
+};
+
+// build a separator label
+const sepLabel = (labelText: string) => {
+    const container = new Container({
+        class: 'control-parent',
+        id: 'sep-container'
+    });
+
+    container.class.add('sep-container');
+
+    const label = new Label({
+        class: 'contol-element-expand',
+        text: labelText
+    });
+
+    container.append(label);
+
+    return container;
+}
+
+// build a data label
+const dataLabel = (parent: Container, labelText: string) => {
+    const container = new Container({
+        class: 'control-parent'
+    });
+
+    const label = new Label({
+        class: 'control-label',
+        text: labelText
+    });
+
+    const value = new Label({
+        class: 'control-element-expand'
+    });
+
+    container.append(label);
+    container.append(value);
+
+    parent.append(container);
+
+    return value;
+};
+
+class DataPanel extends Panel {
+    constructor(events: Events, args = { }) {
+        args = Object.assign(args, {
```

Comment:
Modern JS would be:

args = { ...args,

Reply:
I've updated them all to use the _modern way_ ™